### PR TITLE
Add more fields to apple receipt and add new payment method: Appe Pay

### PR DESCRIPTION
**Pull request by rwhite27**

### CODE REVIEW

File: src/Wallets/PaymentMethods/ApplePay.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+class ApplePay implements PaymentMethodsInterface
+{
+    /**
+     * Receipt Validator Trait
+     */
+    use ReceiptValidatorTrait;
+
+    public ?Users $users = null;
+    public ?string $customer_id = null;
```
Comment (kaioken) [inline src/Wallets/PaymentMethods/ApplePay.php]: @rwhite27 cammelCase

**Review by kaioken — CHANGES_REQUESTED** [review]

File: src/Wallets/PaymentMethods/ApplePay.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+     */
+    public function canCharge() : bool
+    {
+        return true;
+    }
+
+    /**
+     * charge the user
+     * @param float $amount
+     * @return Transactions
```
Comment (arfenis) [inline src/Wallets/PaymentMethods/ApplePay.php]: Seems like the return type is boolean not a Transactions

File: src/Wallets/PaymentMethods/Stripe.php [hunk 3]
```diff
@@ -21,16 +21,16 @@ public function __construct(Users $users)
...
     {
-        return is_null($this->customer_id) ? false : true;
+        return is_null($this->customerId) ? false : true;
     }
 
     /**
      * charge the user
-     * @param float $amount 
+     * @param float $amount
      * @return Transactions
```
Comment (arfenis) [inline src/Wallets/PaymentMethods/Stripe.php]: Same as before

File: tests/integration/Wallets/ApplePayCest.php [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+class ApplePayCest
+{
+    public ChannelsModel $channel;
+
+    /**
+     * Validate Apple Pay receipts
+     *
+     * @param string $receiptData
+     * @param string $source
+     * @return array
```
Comment (arfenis) [inline tests/integration/Wallets/ApplePayCest.php:19]: Update Tests Doc.

File: tests/integration/Wallets/ApplePayCest.php [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+     */
+    public function validatePaymentTest(IntegrationTester $I): void
+    {
+        $receiptData = getenv('ITUNES_RECEIPT_EXAMPLE');
+        $sharedSecret = getenv('ITUNES_STORE_PASS');
```
Comment (arfenis) [inline tests/integration/Wallets/ApplePayCest.php:24]: I didnt saw where did you use this on this test.

**Review by arfenis — CHANGES_REQUESTED** [review]

**Review by arfenis — APPROVED** [review]

**Review by kaioken — APPROVED** [review]

**Pull request merged**